src/ripple/beast/clock/basic_seconds_clock.cpp

```diff
@@ -0,0 +1,103 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of Beast: https://github.com/vinniefalco/Beast
+    Copyright 2021, Howard Hinnant <howard.hinnant@gmail.com>
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/beast/clock/basic_seconds_clock.h>
+
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+namespace beast {
+
+namespace {
+
+// Updates the clock
+class seconds_clock_thread
+{
+    using Clock = basic_seconds_clock::Clock;
+
+    bool stop_;
+    std::mutex mut_;
+    std::condition_variable cv_;
+    std::thread thread_;
+    Clock::time_point tp_;
+
+public:
+    ~seconds_clock_thread();
+    seconds_clock_thread();
+
+    Clock::time_point
+    now();
+
+private:
+    void
+    run();
+};
+
+seconds_clock_thread::~seconds_clock_thread()
+{
+    if (thread_.joinable())
```

Comment:
You can just `assert` this condition here. The dtor is entered only if the ctor exited normally, which only happens if `thread_` is left in a joinable state.

Reply:
Fixed in next commit, thanks.